aten/src/ATen/templates/TypeDerived.cpp

```diff
@@ -65,7 +65,7 @@ Tensor ${Type}::unsafeTensorFromTH(void * th_pointer, bool retain) const {
 std::unique_ptr<Storage> ${Type}::unsafeStorageFromTH(void * th_pointer, bool retain) const {
   if (retain)
     ${THStorage}_retain(${state,} (${THStorage}*) th_pointer);
-  return std::unique_ptr<Storage>(new ${Storage}((${THStorage}*) th_pointer));
+  return std::unique_ptr<Storage>((${THStorage}*) th_pointer);
```

Comment:
This looks wrong. The standard deleter for a `unique_ptr` will call `delete` on the enclosed object. So when `std::unique_ptr<Storage>` becomes dead it will delete it. But what this function used to do is return a `unique_ptr` to Storage, **which is owning only for a single decrement of the reference count**; i.e., when the unique pointer dies, we merely decrement the refcount, and free the storage only if it's dead.

This interface is bad and confusing, but to fix it you would have to first refactor the `storage()` interface to not use unique pointers anymore, and have some sort of wrapper class `Storage` to `StorageImpl` (the same way we have `Tensor` to `TensorImpl`). I would prefer not to have to do it in this PR, but I am not sure if there is a way to fix this without changing the interface of unsafeStorageFromTH

How many use sites of unsafeStorageFromTH are there?

Reply:
There appears to be only two callsites

```
torch/csrc/DynamicTypes.cpp:153:  return type.unsafeStorageFromTH(((THPVoidStorage*)obj)->cdata, true);
torch/csrc/autograd/generated/VariableType.cpp:130:  return baseType->unsafeStorageFromTH(th_pointer, retain);
```